tests/unit/extensions/test_flask.py

```diff
@@ -0,0 +1,192 @@
+                "srv/flask/app/requirements.txt",
+                "srv/flask/app/static",
+            ],
+            "organize": {
+                "app.py": "srv/flask/app/app.py",
+                "node_modules": "srv/flask/app/node_modules",
+                "static": "srv/flask/app/static",
+                "requirements.txt": "srv/flask/app/requirements.txt",
+            },
+            "plugin": "dump",
+            "prime": [
+                "srv/flask/app/app.py",
+                "srv/flask/app/requirements.txt",
+                "srv/flask/app/static",
+            ],
+        },
+    }
+
+
+@pytest.mark.usefixtures("flask_extension")
+def test_flask_extension_overwrite(tmp_path, input_yaml):
+    (tmp_path / "requirements.txt").write_text("flask")
+    (tmp_path / "foobar").touch()
+    (tmp_path / "webapp").mkdir()
+    (tmp_path / "webapp/app.py").touch()
+    (tmp_path / "static").mkdir()
+    (tmp_path / "node_modules").mkdir()
+
+    input_yaml["parts"] = {
+        "flask/install-app": {"prime": ["-srv/flask/app/foobar"]},
+        "flask/dependencies": {"python-requirements": ["requirements-jammy.txt"]},
+    }
+    input_yaml["services"] = {
+        "flask": {
+            "command": "/bin/python3 -m gunicorn --bind 0.0.0.0:8000 webapp.app:app"
+        },
+        "foobar": {
+            "command": "/bin/foobar",
+            "override": "replace",
+        },
+    }
+    applied = extensions.apply_extensions(tmp_path, input_yaml)
+
+    assert applied["services"] == {
+        "flask": {
+            "command": "/bin/python3 -m gunicorn --bind 0.0.0.0:8000 webapp.app:app",
+            "override": "replace",
+            "startup": "enabled",
+            "user": "_daemon_",
+            "working-dir": "/srv/flask/app",
+        },
+        "foobar": {
+            "command": "/bin/foobar",
+            "override": "replace",
+        },
+    }
+    assert applied["parts"]["flask/install-app"]["prime"] == ["-srv/flask/app/foobar"]
+
+    assert applied["parts"]["flask/dependencies"] == {
+        "plugin": "python",
+        "python-packages": ["gunicorn"],
+        "python-requirements": ["requirements.txt", "requirements-jammy.txt"],
+        "source": ".",
+        "stage-packages": ["python3-venv"],
+    }
+
+
+@pytest.mark.usefixtures("flask_extension")
+def test_flask_extension_bare(tmp_path):
+    (tmp_path / "requirements.txt").write_text("flask")
+    input_yaml = {
+        "extensions": ["flask"],
+        "base": "bare",
+        "parts": {"flask/install-app": {"prime": ["-srv/flask/app/.git"]}},
+    }
+    applied = extensions.apply_extensions(tmp_path, input_yaml)
+    assert applied["parts"]["flask/container-processing"] == {
+        "plugin": "nil",
+        "source": ".",
+        "override-build": "mkdir -m 777 ${CRAFT_PART_INSTALL}/tmp",
+    }
+    assert applied["build-base"] == "ubuntu:22.04"
+
+
+@pytest.mark.usefixtures("flask_extension")
+def test_flask_extension_error(tmp_path):
```

Comment:
thanks for testing the error cases! I think we should split this into separate tests. It's a bit of work but it makes each individual test easier to reason about, and lowers the chance that they might interfere with each other in unexpected ways

Reply:
Gotcha, separate them into different test cases with appropriate names, thanks!